### PR TITLE
We need a WYSIWYG copy method

### PR DESCRIPTION
**Pull request by Antecer**
image

### CODE REVIEW

**Comment by Antecer** [comment]
@microsoft-github-policy-service agree

**Comment by Antecer** [comment]
And added WYSIWYG binary search

image

**Comment by Antecer** [comment]
Add this option to set the default copy method
image

File: shared/protocol.ts [hunk 1]
```diff
@@ -179,6 +180,7 @@ export interface DeleteAcceptedMessage {
 }
 
 export const enum CopyFormat {
+	Raw = "Raw",
```
Comment (connor4312) [inline shared/protocol.ts:183]: "Hex Octets" would be a better name for this. Raw implies we're tossing binary onto your clipboard, which we aren't 🙂

File: media/editor/findWidget.tsx [hunk 2]
```diff
@@ -88,7 +88,11 @@ const getSearchQueryOrError = (
 	isBinaryMode: boolean,
 	isRegexp: boolean,
 ): SearchRequestMessage["query"] | string => {
+	const hexPattern = /^\s*([0-9a-fA-F]{2}\s*)+$/;
 	if (isBinaryMode) {
+		if (hexPattern.test(query)) {
+			query = query.replace(/\s/g, "");
+		}
```
Comment (connor4312) [inline media/editor/findWidget.tsx:95]: I think we should just modify `parseHexStringWithPlaceholders` to ignore whitespace in its loop

File: package.json [hunk 3]
```diff
@@ -71,6 +72,22 @@
             "maximum": 512,
             "description": "%hexeditor.columnWidth%"
           },
+          "hexeditor.copyType": {
```
Comment (connor4312) [inline package.json:75]: It'd be cool to also have button in the "Copy As..." pick (`QuickPick.buttons`) to bring the user to this setting.

you can do so with `vscode.commands.executeCommand('workbench.action.openSettings2', { query: '@id:hexeditor.copyType' })`

File: package.nls.json [hunk 4]
```diff
@@ -13,6 +13,7 @@
 	"hexEditor.goToOffset": "Go To Offset",
 	"hexEditor.selectBetweenOffsets": "Select Between Offsets",
 	"hexEditor.copyAs": "Copy As...",
+	"hexeditor.copyType": "Copy Bytes as the selected format",
```
Comment (connor4312) [inline package.nls.json:16]:
```suggestion
	"hexeditor.copyType": "Sets the default format in which bytes are copied",
```

File: media/editor/dataDisplay.tsx [hunk 5]
```diff
@@ -296,7 +297,7 @@ export const DataDisplay: React.FC = () => {
 			select.messageHandler.sendEvent({
 				type: MessageType.DoCopy,
 				selections: ctx.selection.map(r => [r.start, r.end]),
-				format: ctx.focusedElement.char ? CopyFormat.Utf8 : CopyFormat.Base64,
+				format: ctx.focusedElement.char ? CopyFormat.Utf8 : copyType,
```
Comment (connor4312) [inline media/editor/dataDisplay.tsx:300]: The webview doesn't actually need to know about the type the user currently has selected. I would probably replace the `format` property with `text?: boolean` and then map that to whatever the setting value is in the extension side.

File: shared/protocol.ts (hunk 1 above)
Comment (Antecer) [reply shared/protocol.ts:183]: "Hex Octets" looks good.
I just don't know how to describe this structure briefly, so I describe it as what you get is what you see. XD

File: media/editor/findWidget.tsx (hunk 2 above)
Comment (Antecer) [reply media/editor/findWidget.tsx:95]: just need add `\?` in hexPattern, like this `/^\s*([0-9a-fA-F\?]{2}\s*)+$/`

File: package.json (hunk 3 above)
Comment (Antecer) [reply package.json:75]: added by [7d5dfde](https://github.com/microsoft/vscode-hexeditor/pull/540/commits/7d5dfde8c5ba5a1e6900fea6bcba54c6f7dd9d41)

File: media/editor/findWidget.tsx (hunk 2 above)
Comment (Antecer) [reply media/editor/findWidget.tsx]: added by [1083be1](https://github.com/microsoft/vscode-hexeditor/pull/540/commits/1083be1834805b6807298370d35ad526c41814e8)

File: shared/protocol.ts (hunk 1 above)
Comment (Antecer) [reply shared/protocol.ts]: changed by [c3820c4](https://github.com/microsoft/vscode-hexeditor/pull/540/commits/c3820c4ec6a1cbe86dad5eee8b55189d728acdf2)

File: media/editor/dataDisplay.tsx (hunk 5 above)
Comment (Antecer) [reply media/editor/dataDisplay.tsx:300]: i have no idea

**Comment by Antecer** [comment]
Added a button to quickly jump to the CopyType settings：
image

**Review by connor4312 — APPROVED** [review]

**Review by bpasero — APPROVED** [review]

**Pull request merged**